### DISCUSSION TITLE
Subject not recognized w dataPar.json

### DISCUSSION THREAD

**Discussion by YSOskmc1** [ExploreASL/ExploreASL#1284] [Q&A]
Hello,

I am trying to analyze my pCASL data using a subject's T1w scan and perfusion data. I set up my data to have the same format to the ExploreASL test data which I am able to run, but my subject is not being recognized: 
![image](https://user-images.githubusercontent.com/122483786/211905471-f43caa1d-961e-4081-b58d-c4295665e71f.png)

The dataPar.json is identical to the test data, since I am also using only one subject: 

{ "x": [{
	"dataset": {
		"name": "SCS_ASL",
		"subjectRegexp": "^sub-Sub\\d{1}$",
	},
	"settings":	{
		"Quality": 1
	}
}]}

Here are the contents of my raw data folder:

![image](https://user-images.githubusercontent.com/122483786/211905755-c96094b1-356f-4bc6-9081-826ae20bc3aa.png)

...With custom .json files for the T1w and ASL data. Here is the error I always get: 
Warning: Missing x.SUBJECTS field... 
> In xASL_init_DefineStudyData (line 132)
In xASL_init_DataLoading (line 77)
In ExploreASL_Master (line 76)
In ExploreASL (line 14) 
No subjects found...
Please check the sub_regexp in your Data Parameter File.
This should match with the subject folders inside the ROOT folder.
This was C:\Users\yazan\OneDrive\Desktop\SKMC.research\ExploreASL.folderformat ...

Please advise. Thank you.

**Comment by jan-petr** [comment]
Can you try this instead:
"subjectRegexp": ".*Sub\\\\d{1}$",

Careful about the backslashes. in Markdown, backslashes should be escaped and in JSON as well. So in JSON, copy as displayed here - two backslashes.

**Comment by YSOskmc1** [comment]
Thanks Jan Petr, the command is able to run! Unfortunately I run into several other errors regarding ExploreASL recognizing presence of my data: 

Warning: C:\Users\yazan\OneDrive\Desktop\SKMC.research\ExploreASL.folderformat\sub-Sub1 didnt contain a T1w structural image,
skipping...

No ASL found, skipping: C:\Users\yazan\OneDrive\Desktop\SKMC.research\ExploreASL.folderformat\sub-Sub1\ASL_1
Warning: x.Q.readoutDim parameter missing, skipping determining ASL sequence... 
Warning: x.Q.Vendor missing, skipping determining ASL sequence 

3D_spiral sequence detected
No session or subject files found

And more. The data is multi-delay pcASL. But I think the first issue is I'm not sure why the data isn't being recognized...

**Reply by jan-petr** [reply]
That error above typically occurs when it can't manage to convert data from BIDS format to the internal ExploreASL "Legacy" format. It seems to be an error in how ExploreASL is configured, data are saved etc. Below are some hints on what to check.

1. Please use the latest ExploreASL version 1.10.1 (though this shouldn't be the problem in this case).
2. Only run ExploreASL('C:\Users\yazan\OneDrive\Desktop\SKMC.research\ExploreASL.folderformat') command which should convert data from BIDS to ExploreASL format. You will find the data under ExploreASL.folderformat/derivatives/ExploreASL/sub_Sub1 - if NiFTI files are there for T1 and ASL, then you will know that it run correctly.
3. Above, you seem to have your data in C:\Users\yazan\OneDrive\Desktop\SKMC.research\ExploreASL.folderformat\rawdata/sub_Sub1 but the message says something else - is it really in the correct folders like indicated in the JPG?
4. The issue is indeed with not finding the files, not with their content.

See if you can find any help in the points above. If not, please write me an email and we can schedule a Teams meeting.

regards,
Jan
